ql/src/java/org/apache/hadoop/hive/ql/processors/CompileProcessor.java

```diff
@@ -241,14 +254,26 @@ CommandProcessorResponse compile(SessionState ss) throws CommandProcessorExcepti
         out.closeArchiveEntry();
       }
       out.finish();
+      setPosixFilePermissions(testArchive, lockout, false);
```

Comment:
What's the purpose of set the permission here? Could it happen before the write operations above?